shell/src/main/java/alluxio/shell/command/CpCommand.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.shell.command;
+
+import alluxio.AlluxioURI;
+import alluxio.Configuration;
+import alluxio.client.ReadType;
+import alluxio.client.WriteType;
+import alluxio.client.file.FileInStream;
+import alluxio.client.file.FileOutStream;
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.URIStatus;
+import alluxio.client.file.options.CreateFileOptions;
+import alluxio.client.file.options.OpenFileOptions;
+import alluxio.exception.AlluxioException;
+import alluxio.exception.ExceptionMessage;
+import alluxio.exception.FileDoesNotExistException;
+import alluxio.shell.AlluxioShellUtils;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.base.Joiner;
+import com.google.common.io.Closer;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Copies a file or a directory in the Alluxio filesystem.
+ */
+@ThreadSafe
+public final class CpCommand extends AbstractShellCommand {
+
+  /**
+   * @param conf the configuration for Alluxio
+   * @param fs the filesystem of Alluxio
+   */
+  public CpCommand(Configuration conf, FileSystem fs) {
+    super(conf, fs);
+  }
+
+  @Override
+  public String getCommandName() {
+    return "cp";
+  }
+
+  @Override
+  protected int getNumOfArgs() {
+    return 2;
+  }
+
+  @Override
+  public void run(CommandLine cl) throws IOException {
+    String[] args = cl.getArgs();
+    AlluxioURI srcPath = new AlluxioURI(args[0]);
+    AlluxioURI dstPath = new AlluxioURI(args[1]);
+    List<AlluxioURI> srcPaths = AlluxioShellUtils.getAlluxioURIs(mFileSystem, srcPath);
+    if (srcPaths.size() == 0) {
+      throw new IOException(srcPath.getPath() + " does not exist.");
+    }
+
+    if (srcPath.containsWildcard()) {
+      copyWildcard(srcPaths, dstPath);
+    } else {
+      copy(srcPath, dstPath);
+    }
+  }
+
+  /**
+   * Copies a list of files or directories specified by srcPaths to the destination specified by
+   * dstPath. This method is used when the original source path contains wildcards.
+   *
+   * @param srcPaths a list of files or directories in the Alluxio filesystem
+   * @param dstPath the destination in the Alluxio filesystem
+   * @throws IOException
```

Comment:
The benefits are that (1) we won't have to throw non-IO related exceptions as IOExceptions and (2) we don't need exception boilerplate. It seems like the wrapper ought to log any exception that's thrown, not just IOExceptions.

Since the precedent has already been set in other classes we can leave changing the use of IOException for a future PR, but if we keep `@throws IOException` we should include a nonempty javadoc message.